client/state/sharing/keyring/reducer.js

```diff
@@ -28,12 +28,18 @@ export const isFetching = createReducer( false, {
 // Stores the list of available keyring connections
 export const items = createReducer( {}, {
 	[ KEYRING_CONNECTIONS_RECEIVE ]: ( state, { connections } ) => ( { ...keyBy( connections, 'ID' ) } ),
-	[ PUBLICIZE_CONNECTION_CREATE ]: ( state, { ID, siteId } ) => ( { ...state, [ ID ]: {
-		...state[ ID ], sites: [ ...state[ ID ].sites, siteId.toString() ]
-	} } ),
-	[ PUBLICIZE_CONNECTION_DELETE ]: ( state, { ID, siteId } ) => ( { ...state, [ ID ]: {
-		...state[ ID ], sites: without( state[ ID ].sites, siteId.toString() )
-	} } ),
+	[ PUBLICIZE_CONNECTION_CREATE ]: ( state, { connection: { keyring_connection_ID: id, site_ID } } ) => {
```

Comment:
My bad, misread that as breaking up the one-line function body :)

> we don't destructure sites yet it's referenced on the object in the function body anyways

It's referenced on the keyring object that I confusingly named `connection` as well. The original `connection` object doesn't (seem to) get passed to the function once it's destructed.